doc/src/manual/interfaces.md

```diff
@@ -5,6 +5,79 @@ A lot of the power and extensibility in Julia comes from a collection of informa
 receive those functionalities, but they are also able to be used in other methods that are written
 to generically build upon those behaviors.
 
+## [Type Properties](@id man-type-properties)
+
+| Methods to implement              | Default definition           | Brief description                                                                     |
+|:--------------------------------- |:---------------------------- |:------------------------------------------------------------------------------------- |
+| `propertynames(x::ObjType, private::Bool=false)` | `fieldnames(typeof((x))`     | Returns a tuple of the properties (`x.property`) of an object `x`. `private=true` returns fieldnames intended to be kept as private |
+| `getproperty(x::ObjType, s::Symbol)`       | `getfield(x, s)`     | Returns property `s` of `x`. `x.s` calls `getproperty`.  |
+| `setproperty!(x::ObjType, s::Symbol, v)`   | `setfield!(x, s, v)` | Sets property `s` of `x` to `v`. `x.s = v` calls `setproperty!`. |
+
+Sometimes, it is desirable to change how end-user interacts with the fields of an object. 
+Instead of granting direct access to type fields, an extra layer of abstraction between 
+the user and the code can be provided through *type properties*. Properties are what the 
```

Comment:
I wouldn't call them "type properties" since they are functions of the instance, not just the type.

Reply:
Maybe something like "Overloading `object.field`: Properties"